### ISSUE TITLE
Just another short question about data and encrypted files

### ISSUE THREAD

**Issue by Mattel88** [nylira/prism-break#1616]
So I still do not know why VeraCrypt is not recommended on Prism Break since it is open source and there was a security audit which tells us it is safe. 

I'm currently using Apple devices and I look for secure ways to store my files. Since I won't only trust FileVault2 I'm looking for a secure way to store my files. Are the encrypted disk images in disk utilities secure? I'm really not sure what to do actually. Why is encfs recommended and not VeraCrypt? Can I still use Apple encrypted images or are they backdoored? (I really can't imagine there is a backdoor because Apple would lose so much trust and they have an image)

**Comment by hasufell** [comment]
> I'm currently using Apple devices

...your chain of trust is broken. If you can't trust the core of your OS, you can't trust anything that builds on top of it.

**Comment by Mattel88** [comment]
I thought the NSA is only able to read messages or capture chats like FaceMessage and Messages. I didn't read anything about FileVault encryption.

**Comment by hasufell** [comment]
> I thought the NSA

Attackers don't think like developers or even users do. Software is not just an "application" or a composition of modules. Instead, it has a for example a horizontal (for a web page, e.g. GUI -> frontend -> backend -> database) and a vertical (e.g. Ethernet -> IP -> TCP -> HTTP) dimension. This means you don't necessarily target the application directly or as a whole. A DoS attack isn't primarily directed at application code. An url code injection isn't directed at the architecture of your program, but on details of the data flow, which people rarely model explicitly.

You can also think of software in terms of "dependencies" or "environment" and you end up with something like: hardware -> kernel -> libc -> libraries -> your software. If you control what the hardware, kernel and the standard libraries do, you have _complete_ control over what runs on them. We can't really say if that control is utilized in some way, but because we can't there is no trust and we assume the worst.

Even when your favorite software is audited... if you run it in an untrusted environment, you can't be sure how it behaves as a whole. One can intercept system calls, log application behavior and whatever if he has control over the environment.

**Comment by jinformatique** [comment]
@hasufell good explanation! But if you use disk encryption on non secure environment like apple, does it escape mass surveillance?

**Comment by hasufell** [comment]
> But if you use disk encryption on non secure environment like apple, does it escape mass surveillance?

As I said... encryption is already a problem, because of the following reasons:
1. does the user use it correctly? (usually, there is more than one way to do things wrong)
2. are there (unknown) implementation bugs?
3. how strong is the cryptographic algorithm really?

All 3 can be considered attack vectors for an NSA-level player. If you are in an untrusted environment, you open even more attack vectors:
1. does the system monitor me (keylogger, use pattern, which can be used to break into your encrypted files without actually using the previously mentioned attack vectors)
2. does the system manipulate system calls (e.g. writing/reading files can be "intercepted", based on filename/process/use-pattern/whatever)
3. does the system even provide proper randomness?
4. ...probably a few hundred things I didn't think of yet

Attackers use the simplest solution, not the hardest.

If you are building a fortress on sand, how long do you think that will hold?

**Comment by Hillside502** [comment]
@hasufell 
> ...it's wrong to recommend "alternatives" for apple and windows users. It gives a false understanding of what the problem is and fails to educate people about the real problem.

Agreed!

**Comment by alerque** [comment]
What  @hasufell  said. Just to put this in perspective a little bit though remember @jinformatique that we're talking in the context of PRISM-Break there. In other words state sponsored actors are part of the threat model we're reviewing against. If you can't trust your OS then anything at all you do in it is potentially subverted (and this isn't all just an exercise in theory, this stuff actually happens).

This is not to say taking precautions on these platforms is a bad idea. I recommend my friends use password managers, store their sensitive stuff in encrypted vaults, run firewalls that, use full disc encryption, and take other measures even on insecure operating systems. These things protect users against script kiddies, malware, neighborhood electronic thieves, and deceptive co-workers. But that doesn't mean we can recommend them as providing any serious level of protection against truly high tech mass surveillance. The guy who steals your laptop and tries to pillage the hard drives isn't going to have the chops to break into your file vault, but a larger actor could compromise the updates that get sent to your OS and break into your data before it even reaches the disk (I'm generalizing here, the specifics of that kind of attack would actually be different).

> But if you use disk encryption on non secure environment like apple, does it escape mass surveillance?

In short, no. Don't fool yourself. It might escape _other_ threat models and it might dodge _some_ bullets, but the kind of insidious surveillance programs that we at PRISM-Break consider as part of our threat model are not foiled by anything you run if you run it on a compromised platform.

**Comment by jinformatique** [comment]
ok so following this kind of reasoning using hardware with Intel x86 is considered harmful as well.
https://blog.invisiblethings.org/papers/2015/x86_harmful.pdf
New intel proc can be managed remotely.

**Comment by hasufell** [comment]
@jinformatique hardware is another problem yes, but... I'd go so far to say that this is currently out of the scope of prism-break:

> Privacy-aware recommendations to proprietary software.

Starting a proper platform for discussing and recommending hardware is not an easy task I'd say. I don't know if there have been threads about it yet, but I guess the doors are open if people are willing to do the work.

**Comment by strugee** [comment]
@jinformatique that's true. However acquiring hardware that isn't suspicious is very, very difficult. IIRC the FSF does this and as a result has to buy years-old chips. Installing a secure OS is orders of magnitude easier and therefore orders of magnitude more reasonable for us to recommend.

I'm sure there are people in this thread that would _love_ for us to recommend against x86. However we don't because that can of worms is simply too big and too hard - both too hard to find recommendations for and too hard to actually follow those recommendations. (Remember that PRISM Break's goal is to be at least somewhat broadly approachable.)

**Comment by jinformatique** [comment]
That's understandable and I know to have open hardware mainstream, we have to be patient. Work has only begun. Maybe in some years will have better and more secure plateforme.
Just wanted to share some crazy ARM notebook.
https://www.pine64.org/?page_id=3707

**Comment by strugee** [comment]
Closing as discussion isn't going anywhere and there doesn't seem to be any action items in this thread.

**Comment by Atavic** [comment]
BIOS modding for removing the UEFI network stack:
https://github.com/NikolajSchlej/ZeroNights2015
